lib/spack/spack/ci.py

```diff
@@ -426,23 +459,47 @@ def release_jobs(parser, args):
         build_group = ci_cdash['build-group']
         cdash_url = ci_cdash['url']
         cdash_project = ci_cdash['project']
-        proj_enc = urlencode({'project': cdash_project})
-        eq_idx = proj_enc.find('=') + 1
-        cdash_project_enc = proj_enc[eq_idx:]
         cdash_site = ci_cdash['site']
 
-        if args.cdash_credentials:
-            with open(args.cdash_credentials) as fd:
-                cdash_auth_token = fd.read()
-                cdash_auth_token = cdash_auth_token.strip()
+        if 'SPACK_CDASH_AUTH_TOKEN' in os.environ:
+            tty.verbose("Using CDash auth token from environment")
+            cdash_auth_token = os.environ.get('SPACK_CDASH_AUTH_TOKEN')
+
+    # Make sure we use a custom spack if necessary
+    custom_spack_repo = os.environ.get('SPACK_REPO')
+    custom_spack_ref = os.environ.get('SPACK_REF')
+    before_script = None
+    after_script = None
+    if custom_spack_repo:
+        if not custom_spack_ref:
+            custom_spack_ref = 'master'
+        before_script = [
+            'export SPACK_CLONE_LOCATION=$(mktemp -d)',
```

Comment:
I think the `mktemp` is not needed -- is it?  We have the working dir for the runner.

Note that doing it this way limits the control of gitlab admins over where runners store their state, as you'rer always putting this spack in `$TMP`.  Some sites likely won't have room there and will have configured gitlab runners to have their working dirs in good locations to begin with.  so I'd just use `$PWD` instead of `mktemp`.

Reply:
Can we have CLI args for `SPACK_REF` and `SPACK_REPO`, instead of strictly relying on env variables?

I think one of the selling points of this new version of the CI process is that we're demystifying it by cutting down on automatic behavior that depends on things having specific names (e.g.: environment variables and runner tags).  This clarification simplifies the process and empowers users to adapt it to fit their use cases; free from the fear of inadvertently breaking something.

For a more specific use case, consider a site that wants to use this CI process, but wants to stick to a pinned or forked version of Spack pre-installed on their runners.  They might want to clone a recent ref to use these new features, but not clone any version of Spack during the builds.  If they set environment variables to control this pre-ci clone, and *happen* to call them `SPACK_REPO` and `SPACK_REF`, then they'd have to know to unset them before running `spack ci start`.  Using CLI args instead of environment variables makes this behavior clearer and reduces the chance of accidentally having Spack do something they don't want.

Reply:
> I think the `mktemp` is not needed -- is it? We have the working dir for the runner.

@tgamblin I no longer create a temp dir when cloning spack, but rather just do it right inside the environment.  The only issue that caused was in how we push the generated yaml to the downstream repo.  You can see how I'm avoiding pushing the spack directory in `cmd/ci.py`, I'm not too worried about it since it's a temporary workaround until we have child pipelines.

Reply:
> Can we have CLI args for `SPACK_REF` and `SPACK_REPO`, instead of strictly relying on env variables?

@opadron I agree with you,  and I've changed the behavior with regard to `SPACK_REPO` and `SPACK_REF`, so that they are now passed to the generation process as command line arguments.